examples/data_preprocessing/police_killing/data_preprocessing_with_keywords.ipynb

```diff
@@ -0,0 +1,1060 @@
+       "      <td>0</td>\n",
+       "      <td>negative</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>2</td>\n",
+       "      <td>PERSON shooting PERSON shooting TARGET / Chica...</td>\n",
+       "      <td>[police shooting]</td>\n",
+       "      <td>[9]</td>\n",
+       "      <td>0</td>\n",
+       "      <td>negative</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>3</td>\n",
+       "      <td>Scene of Round Lake police shooting Scene of R...</td>\n",
+       "      <td>[police shooting]</td>\n",
+       "      <td>[9]</td>\n",
+       "      <td>0</td>\n",
+       "      <td>negative</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>4</td>\n",
+       "      <td>involved shooting TARGET / Chicago Tribune The...</td>\n",
+       "      <td>[officer killed, officer shooting]</td>\n",
+       "      <td>[24, 29]</td>\n",
+       "      <td>0</td>\n",
+       "      <td>negative</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   index                                            samples  \\\n",
+       "0      0  [ ] TARGET / Chicago Tribune Lake County Major...   \n",
+       "1      1  Round Lake police shooting Round Lake police s...   \n",
+       "2      2  PERSON shooting PERSON shooting TARGET / Chica...   \n",
+       "3      3  Scene of Round Lake police shooting Scene of R...   \n",
+       "4      4  involved shooting TARGET / Chicago Tribune The...   \n",
+       "\n",
+       "                                rules enc_rules  enc_labels    labels  \n",
+       "0                   [police shooting]       [9]           0  negative  \n",
+       "1                   [police shooting]       [9]           0  negative  \n",
+       "2                   [police shooting]       [9]           0  negative  \n",
+       "3                   [police shooting]       [9]           0  negative  \n",
+       "4  [officer killed, officer shooting]  [24, 29]           0  negative  "
+      ]
+     },
+     "execution_count": 20,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "test_data = get_dev_test_df(rule2rule_id, df_test, label_id2label)\n",
+    "test_data.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Converting to sparse matrix\n",
+    "\n",
+    "The Train, Test and Development data that we just stored as Pandas Dataframes should now be converted into a Scipy sparse matrix. The rows of the sparse matrix are the samples and the columns the rules. It is also one-hot-encoded (a cell is 1 if a rule matches a sample.) We inizialize it as an array in the correct size (samples x rules) and set it to 1 if a rule matches the sample. In the end, the array is converted to a sparse matrix."
```

Comment:
```suggestion
    "The train, test, and development data that we just stored as Pandas Dataframes should now be converted into a Scipy sparse matrix. The rows of the sparse matrix are the samples and the columns are the rules. The rows of this matrix are one-hot-encoded vectors (i.e., a cell is 1 if the corresponding rule matches the corresponding sample, 0 otherwise.) We initialize it as an array in the correct size (samples x rules), fill it with 1s and 0s, and convert it to a sparse matrix at the end."
```

Reply:
in general, in z matric the rows are not necessarily one-hot-encoded vectors (since several rules can match in one sample). Or I got it wrong?

Reply:
You're right. How would I call it correctly?